src/SqlAst/MinMaxReturnTypeExtension.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace staabm\PHPStanDba\SqlAst;
+
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use SqlFtw\Sql\Expression\BuiltInFunction;
+use SqlFtw\Sql\Expression\FunctionCall;
+
+final class MinMaxReturnTypeExtension implements QueryFunctionReturnTypeExtension
+{
+    public function isFunctionSupported(FunctionCall $expression): bool
+    {
+        return \in_array($expression->getFunction()->getName(), [BuiltInFunction::MIN, BuiltInFunction::MAX], true);
+    }
+
+    public function getReturnType(FunctionCall $expression, QueryScope $scope): ?Type
+    {
+        $args = $expression->getArguments();
+
+        if (1 !== \count($args)) {
+            return null;
+        }
+
+        $argType = $scope->getType($args[0]);
+
+        if ($argType->isSuperTypeOf(new StringType())->yes()) {
```

Comment:
isString()->yes()

Reply:
I believe that `isSuperTypeOf` is necessary here. For example, consider `SELECT max(ifnull(email, adaid)) as max from ada`. The argument to `max` is `string|int`, and if there are _any_ string inputs, the output is always converted to a string (even if the element it selects came from `adaid`).